File: vandalizer-next/frontend/src/types/automation.ts
export type TriggerType = 'folder_watch' | 'm365_intake' | 'api' | 'schedule'
export type ActionType = 'workflow' | 'extraction' | 'task'

export interface Automation {
  id: string
  name: string
  description: string | null
  enabled: boolean
  trigger_type: TriggerType
  trigger_config: Record<string, unknown>
  action_type: ActionType
  action_id: string | null
  user_id: string
  team_id: string | null
  shared_with_team: boolean
  space: string | null
  created_at: string
  updated_at: string
}

File: vandalizer-next/backend/app/models/automation.py
"""Automation model for trigger-based workflow execution."""

import datetime
from typing import Optional

from beanie import Document


class Automation(Document):
    """An automation that links a trigger to an action."""

    name: str
    description: Optional[str] = None
    enabled: bool = False
    trigger_type: str = "folder_watch"  # folder_watch | m365_intake | api | schedule
    trigger_config: dict = {}
    action_type: str = "workflow"  # workflow | extraction | task
    action_id: Optional[str] = None
    user_id: str
    team_id: Optional[str] = None
    shared_with_team: bool = False
    space: Optional[str] = None
    created_at: datetime.datetime = datetime.datetime.now(tz=datetime.timezone.utc)
    updated_at: datetime.datetime = datetime.datetime.now(tz=datetime.timezone.utc)

    class Settings:
        name = "automation"
